main.js

```diff
@@ -12,23 +13,59 @@ document.addEventListener('keydown', arrows);
 function arrows(event) {
   if (event.key === 'ArrowUp') {
     $car.setAttribute('class', 'car-image up-arrow');
+    data.carDirection = 'North';
   } else if (event.key === 'ArrowRight') {
     $car.setAttribute('class', 'car-image right-arrow');
+    data.carDirection = 'East';
   } else if (event.key === 'ArrowDown') {
     $car.setAttribute('class', 'car-image down-arrow');
+    data.carDirection = 'South';
   } else if (event.key === 'ArrowLeft') {
     $car.setAttribute('class', 'car-image left-arrow');
+    data.carDirection = 'West';
   }
-  if (event.key === ' ' && data.carMoving === false) {
 
+  if (event.key === ' ' && data.carMoving === false) {
     data.carMoving = true;
+  } else if (event.key === ' ' && data.carMoving === true) {
+    clearInterval(count);
+    data.carMoving = false;
+  }
+  if (data.carMoving === true && (event.key === 'ArrowRight' || data.carDirection === 'East')) {
+    distance = data.xAxis;
+    clearInterval(count);
     count = setInterval(function () {
+      $car.style.top = data.yAxis + 'px';
       $car.style.left = (parseInt(data.xAxis) + parseInt(5)) + 'px';
       distance = distance + 5;
       data.xAxis = distance;
     }, 16);
-  } else if (event.key === ' ' && data.carMoving === true) {
+  } else if (data.carMoving === true && (event.key === 'ArrowLeft' || data.carDirection === 'West')) {
     clearInterval(count);
-    data.carMoving = false;
+    distance = data.xAxis;
+    count = setInterval(function () {
+      $car.style.top = data.yAxis + 'px';
+      $car.style.left = (parseInt(data.xAxis) - parseInt(5)) + 'px';
+      distance = distance - 5;
+      data.xAxis = distance;
+    }, 16);
```

Comment:
You should make these anonymous functions into named functions to reduce code bulk